### PR TITLE
Placeholder for disabled channel settings

### LINKED ISSUE #246: Channel settings are not well ordered in HTML
## Description

Channel settings elements appear out of logical order in HTML, making tab-navigating confusing. Elements are grouped like this, in the order marked:

image

This is especially problematic given that the last element of box 3 controls whether box 2 is accessible. A user that expects to open the settings menu with that control then tab into it must instead counterintuitively backtrack past the visibility checkboxes to get at the *end* of the channel settings.

## Expected Behavior

Elements should appear in the correct order and navigating with the keyboard into the advanced channel settings should be intuitive.

## Reproduction

1. Click the "open channel settings" button on any channel.
2. Try to access and adjust that channel's settings with the keyboard (Tab, Shift+Tab, etc.)

## Environment

Firefox, but this would appear in any environment.

### PR DESCRIPTION
**Pull request by frasercl**
Advances #80: adds placeholders for volume/surface settings when a channel's volume/surface is disabled, and allows opening channel settings when the channel is completely invisible.

Also does some related cobweb-clearing and maintenance:
- Converts `ColorPicker` to a function component
- Following discussion with @lynwilhelm, adds a tighter, lighter border around color picker swatches
- Closes #246: puts elements within a channel row in the expected order
- Cleans out a lot of unused or redundant css and css-in-js

Design: https://www.figma.com/design/yGTPga2j961NKfQT7xTird/2023-Sept-Settings-Redesign?node-id=1-2&t=ndIJ40DdmIMNtKgC-0

### Screenshots

| Design | This branch |
| ----- | ----- |
| image | image |

### CODE REVIEW

**Review by toloudis — APPROVED** [review]
I love the net negative lines of code too.

File: src/aics-image-viewer/components/ChannelsWidgetRow/index.tsx [hunk 1]
```diff
@@ -177,52 +161,45 @@ const ChannelsWidgetRow: React.FC<ChannelsWidgetRowProps> = (props: ChannelsWidg
...
 
-  const rowClass = controlsOpen ? "row-card" : "row-card controls-closed";
+  const rowClass = controlsOpen ? "channel-row" : "channel-row controls-closed";
   return (
-    <List.Item key={index} className={rowClass} extra={renderActions()}>
+    <List.Item key={index} className={rowClass}>
       <List.Item.Meta title={<span style={STYLES.channelName}>{props.name}</span>} avatar={createColorPicker()} />
-      {controlsOpen && renderControls()}
+      {visibilityControls}
+      {controlsOpen && <div style={{ width: "100%" }}>{renderControls()}</div>}
```
Comment (interim17) [inline src/aics-image-viewer/components/ChannelsWidgetRow/index.tsx:191]: I like how clean this is, but have been guided away from inline styling in the past.

This is not a request for change, but just curious your thought process in deciding to use inline here.

**Review by interim17 — APPROVED** [review]
Nice work :)

File: src/aics-image-viewer/components/ChannelsWidgetRow/index.tsx (hunk 1 above)
Comment (frasercl) [reply src/aics-image-viewer/components/ChannelsWidgetRow/index.tsx:191]: This codebase is on the older side and some components have an odd pattern of doing styling with big JS objects laid out like stylesheets at the bottom of .tsx files. (I ended up having to add a type for these objects, `Styles`, when I converted the project to TS.) Though I don't take a super hard line on inline styling, I agree it's not great practice and am not really pleased to be doing it here. I just didn't try to break this file all the way out of that pattern... for now. I think this particular case was a sort of half-measure for a style record that I was able to trim down pretty far but not completely get rid of.

**Pull request merged**